google-cloud-container/src/test/java/com/google/cloud/container/v1/it/ITSystemTest.java

```diff
@@ -0,0 +1,186 @@
+            .setSelfLink(NODE_POOL_SEL_LINK)
+            .setStatusMessage(STATUS_MESSAGE)
+            .build();
+    Thread.sleep(TimeUnit.MINUTES.toMillis(5));
+    client.createNodePool(PROJECT_ID, ZONE, CLUSTER_NAME, nodePool);
+    LOG.info(String.format("%s node pool created successfully.", NODE_POOL_NAME));
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    Thread.sleep(TimeUnit.MINUTES.toMillis(5));
+    client.deleteCluster(PROJECT_ID, ZONE, CLUSTER_NAME);
+    LOG.info(String.format("%s cluster deleted successfully.", CLUSTER_NAME));
+
+    client.close();
+  }
+
+  @Test
+  public void getClusterTest() {
+    Cluster cluster = client.getCluster(PROJECT_ID, ZONE, CLUSTER_NAME);
+    assertEquals(CLUSTER_NAME, cluster.getName());
+    assertEquals(DETAIL, cluster.getDescription());
+    assertEquals(ZONE, cluster.getZone());
+    assertEquals(SELF_LINK, cluster.getSelfLink());
+    assertEquals(NETWORK, cluster.getNetwork());
+    assertEquals(INITIAL_NODE_COUNT, cluster.getInitialNodeCount());
+    assertEquals(LOGGING_SERVICE, cluster.getLoggingService());
+    assertEquals(LOGGING_SERVICE, cluster.getLoggingService());
+  }
+
+  @Test
+  public void listClusterTest() {
+    ListClustersResponse clustersResponse = client.listClusters(PROJECT_ID, ZONE);
+    List<Cluster> clusters = clustersResponse.getClustersList();
+    for (Cluster cluster : clusters) {
+      if (CLUSTER_NAME.equals(cluster.getName())) {
+        assertEquals(CLUSTER_NAME, cluster.getName());
+        assertEquals(DETAIL, cluster.getDescription());
+        assertEquals(ZONE, cluster.getZone());
+        assertEquals(SELF_LINK, cluster.getSelfLink());
+        assertEquals(NETWORK, cluster.getNetwork());
+        assertEquals(INITIAL_NODE_COUNT, cluster.getInitialNodeCount());
+        assertEquals(LOGGING_SERVICE, cluster.getLoggingService());
+        assertEquals(LOGGING_SERVICE, cluster.getLoggingService());
+      }
+    }
+  }
+
+  @Test
+  public void getOperationTest() {
+    Operation response = client.getOperation(PROJECT_ID, ZONE, operation.getName());
+    assertEquals(operation.getName(), response.getName());
+    assertEquals(ZONE, response.getZone());
+  }
+
+  @Test
+  public void listOperationsTest() {
+    ListOperationsResponse listOperations = client.listOperations(PROJECT_ID, ZONE);
+    List<Operation> operations = listOperations.getOperationsList();
+    for (Operation operation1 : operations) {
+      if (operation1.getName().equals(operation.getName())) {
+        assertEquals(operation.getName(), operation1.getName());
+        assertEquals(ZONE, operation1.getZone());
+      }
+    }
+  }
+
+  @Test
+  public void getServerConfigTest() {
+    ServerConfig config = client.getServerConfig(PROJECT_ID, ZONE);
+    assertEquals("COS", config.getDefaultImageType());
+    assertEquals(
+        "1.14.7-gke.10",
+        config.getValidNodeVersions(ServerConfig.VALID_MASTER_VERSIONS_FIELD_NUMBER));
```

Comment:
What is the timeline of this release version getting incremented? I mean Are we sure that server ValidNodeVersion would not increment in long time?

Reply:
not sure but might be it will take long period of time :)